pkg/agent/handler/handlers.go

```diff
@@ -107,3 +117,29 @@ func (c categoryHandler) handle(action proto.Event_Type, resource *apiv1.Resourc
 
 	return nil
 }
+
+type agentResourceHandler struct {
+	agentResourceManager resource.Manager
+}
+
+// NewAgentResourceHandler - creates a Handler for Agent resources
+func NewAgentResourceHandler(agentResourceManager resource.Manager) Handler {
+	return &agentResourceHandler{
+		agentResourceManager: agentResourceManager,
+	}
+}
+
+func (h *agentResourceHandler) Handle(action proto.Event_Type, resource *v1.ResourceInstance) error {
+	if h.agentResourceManager != nil && action == proto.Event_UPDATED {
+		kind := resource.Kind
+		switch kind {
+		case discoveryAgent:
+			fallthrough
+		case traceabilityAgent:
+			fallthrough
+		case governanceAgent:
+			h.agentResourceManager.SetAgentResource(resource)
```

Comment:
Do other resource kinds come in here?  Is this switch necessary?

Reply:
The handler gets invoked for all resource events received by the watch client, so this handler does the check for agent resource kind to process them further